src/dmd/dcast.d

```diff
@@ -3302,6 +3302,21 @@ LmodCompare:
             e2 = tmp;
         }
     }
+    // constness1(T)[][] op constness2(T)[][]: maybe we can make them both const?
+    else if (t1.ty == Tarray && t2.ty == Tarray // two dynamic arrays
+        && t1.constOf().equals(t2.constOf()) // and they're unifiable as const
+        && (!t1.equals(t1.constOf()) || !t2.equals(t2.constOf())) // but they're not both const yet
+        && e1.implicitConvTo(t1.constOf()) >= MATCH.constant // but we can make them const
+        && e2.implicitConvTo(t2.constOf()) >= MATCH.constant)
```

Comment:
That looks like an extremely specific check. What about other kind of indirections ? (ptr, `ref`)
```D
    const ca = "a".ptr;
    if ([[ca]] == [["a".ptr]]) {}
```

Does this compile after your change ? Should it ?

Reply:
Looking at the rest of the function, admittedly this check is very specific but it's not all that much more specific than the other ten checks already chained in there.

I wanted to specifically nail this one case, because I am still very unconfident about the general case.

The pasted code compiles, as it should. Is there any reason it shouldn't?